Fix UncountedCallArgs errors for code generated by generate-inspector-protocol-bindings.py

fred-wang opened the pull request:
#### 1a60c58e0682f10b64c0d640938adeaffdce4b4f
<pre>
Fix UncountedCallArgs errors for code generated by generate-inspector-protocol-bindings.py
<a href="https://bugs.webkit.org/show_bug.cgi?id=300309">https://bugs.webkit.org/show_bug.cgi?id=300309</a>

Reviewed by Devin Rousso.

m_result is a RefPtr member but per [1] we need either it to be a smart
pointer on the stack or to be a const. The latter is not possible
because of [2].

[1] <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#mark-data-members-that-are-smart-pointers-as-const-if-they-never-get-reassigned">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#mark-data-members-that-are-smart-pointers-as-const-if-they-never-get-reassigned</a>
[2] <a href="https://searchfox.org/wubkat/rev/dd05eb230972a07f0240b959e6e22ca685ef3445/Source/JavaScriptCore/inspector/scripts/codegen/cpp_generator_templates.py#224">https://searchfox.org/wubkat/rev/dd05eb230972a07f0240b959e6e22ca685ef3445/Source/JavaScriptCore/inspector/scripts/codegen/cpp_generator_templates.py#224</a>

* Source/JavaScriptCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations: Update expectations.
* Source/JavaScriptCore/inspector/scripts/codegen/generate_cpp_protocol_types_header.py: Store smart pointer for m_result on the stack.
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-async-attribute.json-result: Regenerate expectation.
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-optional-call-return-parameters.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/definitions-with-mac-platform.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-exposed-as-other-name.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/events-with-optional-parameters.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-command-targetTypes-value.json-error: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-domain-targetTypes-value.json-error: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-event-targetTypes-value.json-error: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/generate-domains-with-feature-guards.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/shadowed-optional-type-setters.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-object-type.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-object-type.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-requiring-runtime-casts.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-with-open-parameters.json-result: Ditto.
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations: Update expectations.

Canonical link: <a href="https://commits.webkit.org/301256@main">https://commits.webkit.org/301256@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/37512d395e96c93d870baa5d4d41721b3e6ce82e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125231 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44898 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35637 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132082 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77194 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45589 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53517 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95341 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63347 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128185 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36400 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111988 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75912 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35297 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30156 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75560 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/117322 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106167 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30381 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134763 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123749 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52040 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39877 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103810 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52475 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108209 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103577 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26412 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48932 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27221 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49126 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51932 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57711 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/156772 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51294 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39255 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54650 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52988 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/JavaScriptCore/inspector/scripts/codegen/generate_cpp_protocol_types_header.py:
```diff
@@ -342,7 +342,8 @@ def _generate_builder_setter_for_member(self, type_member, domain):
         lines.append('            static_assert(!(STATE & %(camelName)sSet), "property %(memberKey)s already set");' % setter_args)
-        lines.append('            m_result->%(setter)s("%(memberKey)s"_s, %(memberValue)s);' % setter_args)
+        lines.append('            RefPtr protectedResult = m_result;')
+        lines.append('            protectedResult->%(setter)s("%(memberKey)s"_s, %(memberValue)s);' % setter_args)
```
I think you're supposed to rebaseline the tests when you modify the generator:
```
run-inspector-generator-tests --reset-results
```

Also, can we avoid the extra local variable? Maybe `Ref { * m_result }->`

dcrousso commented on Source/JavaScriptCore/inspector/scripts/codegen/generate_cpp_protocol_types_header.py:
```diff
@@ -342,7 +342,7 @@ def _generate_builder_setter_for_member(self, type_member, domain):
         lines.append('        {')
         lines.append('            static_assert(!(STATE & %(camelName)sSet), "property %(memberKey)s already set");' % setter_args)
-        lines.append('            m_result->%(setter)s("%(memberKey)s"_s, %(memberValue)s);' % setter_args)
+        lines.append('            Ref { * m_result }->%(setter)s("%(memberKey)s"_s, %(memberValue)s);' % setter_args)
```
there's another line like this just below.  should it also be updated?
```diff
-m_result->setValue("%(memberKey)s"_s, JSON::Value::null());' % setter_args)
+Ref { *m_result }->setValue("%(memberKey)s"_s, JSON::Value::null());' % setter_args)
```

dcrousso reviewed (approved):
neat!

fred-wang committed:
Fix UncountedCallArgs errors for code generated by generate-inspector-protocol-bindings.py
https://bugs.webkit.org/show_bug.cgi?id=300309

Reviewed by Devin Rousso.

m_result is a RefPtr member but per [1] we need either it to be a smart
pointer on the stack or to be a const. The latter is not possible
because of [2].

[1] https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#mark-data-members-that-are-smart-pointers-as-const-if-they-never-get-reassigned
[2] https://searchfox.org/wubkat/rev/dd05eb230972a07f0240b959e6e22ca685ef3445/Source/JavaScriptCore/inspector/scripts/codegen/cpp_generator_templates.py#224

* Source/JavaScriptCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations: Update expectations.
* Source/JavaScriptCore/inspector/scripts/codegen/generate_cpp_protocol_types_header.py: Store smart pointer for m_result on the stack.
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-async-attribute.json-result: Regenerate expectation.
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-optional-call-return-parameters.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/definitions-with-mac-platform.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-exposed-as-other-name.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/events-with-optional-parameters.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-command-targetTypes-value.json-error: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-domain-targetTypes-value.json-error: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-event-targetTypes-value.json-error: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/generate-domains-with-feature-guards.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/shadowed-optional-type-setters.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-object-type.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-object-type.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-requiring-runtime-casts.json-result: Ditto.
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-with-open-parameters.json-result: Ditto.
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations: Update expectations.

Canonical link: https://commits.webkit.org/301256@main
